ros2node/ros2node/verb/get_type_description.py

```diff
@@ -0,0 +1,156 @@
+    print()
+    print('Referenced Type Descriptions:')
+    for rtd in td.referenced_type_descriptions:
+        print_individual_type_description(rtd, indent=2)
+
+
+def discover_hash_for_type(node, remote_node_name, type_name):
+    # TODO(emersonknapp) get hashes from names_and_types when that is implemented
+    ns = type_name.split('/')[1]
+    if ns != 'msg':
+        raise RuntimeError(
+            f'Currently cannot auto-discover hashes for "{ns}", only "msg". '
+            'Please provide type_hash value to command.')
+    remote_node = parse_node_name(remote_node_name)
+    nt = node.get_publisher_names_and_types_by_node(remote_node.name, remote_node.namespace)
+    nt += node.get_subscriber_names_and_types_by_node(remote_node.name, remote_node.namespace)
+    discover_topic = None
+    for topic, types in nt:
+        if type_name in types:
+            discover_topic = topic
+    if not discover_topic:
+        raise RuntimeError(
+            f'Could not find type "{type_name}" advertised as a topic type on node '
+            f'"{remote_node_name}"')
+
+    endpoint_infos = node.get_publishers_info_by_topic(discover_topic)
+    endpoint_infos += node.get_subscriptions_info_by_topic(discover_topic)
+    for endpoint_info in endpoint_infos:
+        if (
+            endpoint_info.node_name == remote_node.name and
+            endpoint_info.node_namespace == remote_node.namespace and
+            endpoint_info.topic_type == type_name
+        ):
+            return str(endpoint_info.topic_type_hash)
+    raise RuntimeError(
+        f'Cound not find endpoint type hash for topic "{discover_topic}"')
+
+
+class GetTypeDescriptionVerb(VerbExtension):
+    """Output information about a node."""
+
+    def add_arguments(self, parser, cli_name):
+        add_arguments(parser)
+        argument = parser.add_argument(
+            'node_name',
+            help='Node name to request information from')
+        argument.completer = NodeNameCompleter()
+        parser.add_argument(
+            'type_name',
+            help='Interface type to get description of')
+        parser.add_argument(
+            'type_hash',
+            default=None,
+            nargs='?',
+            help='Hash string of the type. If not provided, will try to determine automatically.')
+        parser.add_argument(
+            '--include-sources', action='store_true',
+            help='Fetch and display the raw source text as well')
+
+    def main(self, *, args):
+        service_name = f'{args.node_name}/get_type_description'
+
+        rclpy.init()
+        node = rclpy.create_node(f'{NODE_NAME_PREFIX}_td_requester_{os.getpid()}')
+        cli = node.create_client(GetTypeDescription, '/talker/get_type_description')
+        if not cli.service_is_ready():
+            print('Waiting for service to become available...')
+            if not cli.wait_for_service(timeout_sec=5.0):
+                raise RuntimeError(f'Service {service_name} not found')
```

Comment:
how about waiting until service is ready, unless user sends the signal to stop? i would have timeout option instead of static timeout that user cannot even see how long it waits.

```suggestion
            cli.wait_for_service()
```